src/interfaces/root-state-interface.ts

```diff
@@ -0,0 +1,14 @@
+import { CurrentUserIF, CurrentAccountIF } from '@/interfaces'
+
+/** The ROOT state model interface. */
+export interface RootStateIF {
+  account: {
+    currentUser: CurrentUserIF,
+    currentAccount: CurrentAccountIF
+  },
+  auth: {
+    token: string,
+    idToken: string,
+    refreshToken: string
+  }
```

Comment:
Should the other 2 properties (kcGuid and loginSource) be added here?

Why did you decide to embed the auth interface here? Or conversely, why not move the other interfaces here (even if separate from RootStateIF?)

Reply:
Yes, it's the root state, but since we're not using the 'notification' and 'status' objects I didn't include them, but I can add them if you wish.  The kcGuid and loginSource are available via the CurrentUserIF.  Since the CurrentUserIF (Keycloak) is the authoritative source I thought it would be best to leave them there.  Please push back if you disagree.  There are many ways to skin this 🐱

Reply:
Poor cat.

I was thinking that all the other properties under root (ie, our original store instance) weren't declared here. Should they be?

![image](https://user-images.githubusercontent.com/10002889/218618244-6a7c352a-b6b7-45bf-b28f-ce0517b5381b.png)

I was expecting an object named "root" to include the properties on the right hand side.

Reply:
> I was expecting an object named "root" to include the properties on the right hand side.

To be honest, I don't know what's appropriate here.  It's likely possible to access our state via the 'rootState' parameter, but that definitely not something we want to encourage.  Instead, we should access our `state` via the first parameter which would be appropriately typed.  Like this:
```
getSomething(state: InterfaceIF, getters, rootState) {
      return [ 'blah', 'blah', 'blah']
    }
```

Reply:
Ah, fair enough. Then leave it as is and maybe add a comment summarizing the above? Thx.